lib/cybersource_rest_client/utilities/capture_context/capture_context_parsing_utility.rb

```diff
@@ -0,0 +1,117 @@
+require_relative '../../../AuthenticationSDK/util/JWT/JWTExceptions'
+require_relative '../../../AuthenticationSDK/util/Cache'
+require_relative 'public_key_fetcher'
+
+module CyberSource
+  module Utilities
+    module CaptureContext
+      # CaptureContextParser Class
+      #
+      # Parses and validates capture context JWT responses from CyberSource.
+      # Provides functionality to parse JWT tokens and optionally verify their
+      # signatures using public keys fetched from the CyberSource API.
+      #
+      # @category Class
+      # @package  CyberSource::Authentication::Util::CaptureContext
+      # @author   CyberSource
+      class CaptureContextParser
+        class << self
+          # Parses a capture context JWT response and optionally verifies its signature
+          #
+          # @param jwt_value [String] The JWT token to parse
+          # @param merchant_config [Object] The merchant configuration object
+          # @param verify_jwt [Boolean] Whether to verify the JWT signature (default: true)
+          #
+          # @return [Hash] The parsed JWT payload
+          # @raise [CyberSource::Authentication::Util::JWT::InvalidJwtException] If JWT is invalid
+          # @raise [CyberSource::Authentication::Util::JWT::JwtSignatureValidationException] If signature verification fails
+          # @raise [ArgumentError] If required parameters are missing
+          # @raise [StandardError] For other errors during processing
+          #
+          # @example Parse without verification
+          #   payload = CaptureContextParser.parse_capture_context_response(jwt_token, config, false)
+          #
+          # @example Parse with verification (default)
+          #   payload = CaptureContextParser.parse_capture_context_response(jwt_token, config)
+          def parse_capture_context_response(jwt_value, merchant_config, verify_jwt = true)
+            # Always validate JWT value first
+            if jwt_value.nil? || jwt_value.empty?
+              raise CyberSource::Authentication::Util::JWT::InvalidJwtException.new('JWT value is null or undefined')
+            end
+            
+            # Validate verification requirements before parsing
+            if verify_jwt
+              raise ArgumentError, 'merchantConfig is required' unless merchant_config
+              
+              run_environment = merchant_config.runEnvironment
+              raise ArgumentError, 'Run environment not found in merchant config' unless run_environment
+            end
+            
+            # Parse JWT (actual operation)
+            parsed_jwt = CyberSource::Authentication::Util::JWT::JWTUtility.parse(jwt_value)
+            
+            # Return early if no verification needed
+            return parsed_jwt[:payload] unless verify_jwt
+            
+            # Validate kid exists in parsed JWT
+            kid = parsed_jwt[:header]['kid']
+            unless kid
```

Comment:
Please check this line. The `unless` keyword checks the existence of the variable `kid`, not if it is not falsy (actual Ruby word). "falsy" means `nil` or `false`